src/canvas.ts

```diff
@@ -1,13 +1,16 @@
 import { config } from "./config";
 import { Creature } from "./creatures/creature";
 import { Vector2 } from "./vector2";
+import { Metric } from "./metric";
 
 export class Canvas {
+    private readonly metric: Metric;
     private cameraPosition: Vector2;
     private canvas: HTMLCanvasElement;
     private ctx: CanvasRenderingContext2D;
 
     constructor(canvasElement: HTMLCanvasElement) {
+        this.metric = Metric.getMetric();

```

Comment:
See comment on Metric class name. TL;DR: I think we should name this more specifically 'frameCounter' or similar.